src/uct/ib/rc/base/rc_iface.h

```diff
@@ -493,4 +496,34 @@ uct_rc_iface_fence_relaxed_order(uct_iface_h tl_iface)
 
     return uct_rc_iface_fence(tl_iface, 0);
 }
+
+#if UCS_ENABLE_ASSERT
+static UCS_F_ALWAYS_INLINE int
+uct_rc_iface_check_pending(uct_rc_iface_t *iface, ucs_arbiter_group_t *arb_group)
+{
+    return (iface->tx.in_pending || ucs_arbiter_group_is_empty(arb_group));
```

Comment:
maybe do ucs_assert inside this function? would remove some code, and can remove `#if` from line 500